ambry-cloud/src/test/java/com/github/ambry/cloud/LatchBasedInMemoryCloudDestination.java

```diff
@@ -149,10 +172,22 @@ public void downloadBlob(BlobId blobId, OutputStream outputStream) throws CloudS
 
   @Override
   public boolean deleteBlob(BlobId blobId, long deletionTime, short lifeVerion,
-      CloudUpdateValidator cloudUpdateValidator) {
+      CloudUpdateValidator cloudUpdateValidator) throws CloudStorageException {
+    StoreErrorCodes serverError =
+        hardError != null ? hardError : serverErrors.size() > 0 ? serverErrors.poll() : null;
+    if (serverError != null) {
+      throw new CloudStorageException("deleteBlob simulated error",
+          new StoreException("deleteBlob simulated error", serverError));
+    }
+
     if (!map.containsKey(blobId)) {
       return false;
     }
+    // The lifeVersion from message info is -1 when the undelete method is invoked by frontend request, we have to
+    // get the legit lifeVersion before we can write undelete record to log segment.
+    if (!MessageInfo.hasLifeVersion(lifeVerion)) {
+      lifeVerion = map.get(blobId).getFirst().getLifeVersion();
```

Comment:
nit let's change lifeVerion to lifeVersion

Reply:
Fixed. Thanks,